packages/neo-one-smart-contract-lib/src/token/SecondaryNEP5Token.ts

```diff
@@ -0,0 +1,171 @@
+// tslint:disable readonly-keyword readonly-array no-object-mutation strict-boolean-expressions
+import {
+  Address,
+  constant,
+  Contract,
+  createEventNotifier,
+  Fixed,
+  MapStorage,
+  SmartContract,
+} from '@neo-one/smart-contract';
+
+interface TokenPayableContract {
+  readonly approveReceiveTransfer: (from: Address, amount: Fixed<0>, asset: Address) => boolean;
+  readonly onRevokeSendTransfer: (from: Address, amount: Fixed<0>, asset: Address) => void;
+}
+
+export function SecondaryNEP5Token<TBase extends Constructor<SmartContract>>(Base: TBase) {
```

Comment:
I must be missing something - why is this called `SecondaryNEP5Token`? How does this differ from the old `NEP5Token` mixin?

Reply:
This is a work in progress.  Sorry, should have commented in the notes.